includes/chart.js

```diff
@@ -0,0 +1,330 @@
+                return;
+            }
+
+            var group = document.createElementNS("http://www.w3.org/2000/svg", "g");
+            var newNode;
+            var x = template_node.x + rowX;
+            var y = template_node.y + rowY + (i * rowHeight);
+            if (template_node.shape === 'polygon') {
+                // Create a polygon for the node
+                newNode = document.createElementNS("http://www.w3.org/2000/svg", "polygon");
+                newNode.setAttribute("points", template_node.points);
+                // Calculate centroid of the polygon
+                var centroid = calculatePolygonCentroid(newNode);
+
+                // Calculate translation needed to move the centroid to (x, y)
+                var translationX = x - centroid.x;
+                var translationY = y - centroid.y;
+
+                // Apply translation to the polygon
+                newNode.setAttribute("transform", "translate(" + translationX + "," + translationY + ")");
+            } else if (template_node.shape === 'circle') {
+                /// Create an ellipse for the node
+                newNode = document.createElementNS("http://www.w3.org/2000/svg", "ellipse");
+                newNode.setAttribute("rx", template_node.radiusX); // Use the horizontal radius attribute
+                newNode.setAttribute("ry", template_node.radiusY); // Use the vertical radius attribute
+                newNode.setAttribute("cx", x);
+                newNode.setAttribute("cy", y);
+            } else if (template_node.shape === 'rect') {
+                // Create a rectangle for the node
+                newNode = document.createElementNS("http://www.w3.org/2000/svg", "rect");
+                newNode.setAttribute("x", x);
+                newNode.setAttribute("y", y);
+                newNode.setAttribute("width", template_node.width); // Use the width attribute
+                newNode.setAttribute("height", template_node.height); // Use the height attribute
+                if (template_node.rx && template_node.ry) {
+                    newNode.setAttribute("rx", template_node.rx); // Set horizontal radius for rounded corners
+                    newNode.setAttribute("ry", template_node.ry); // Set vertical radius for rounded corners
+                }
+            }
+
+            newNode.setAttribute("id", `${row.id}_${template_node.id}`);
+            newNode.setAttribute("fill", colors[row.statementType]);
+            newNode.setAttribute("stroke", "black")
+
+            // Add text inside the shape
+            var text = document.createElementNS("http://www.w3.org/2000/svg", "text");
+            text.textContent = row[template_node.type];
+            text.setAttribute("fill", "white");
+            text.setAttribute("font-size", "12px");
+            text.setAttribute("font-family", "calibri");
+            text.setAttribute("text-anchor", "middle"); // Center align the text
+
+            var textX = x, textY = y;
```

Comment:
`textX` and `textY` could be more descriptive, I don't know what they mean.

Reply:
They are the x and y coordinate of where to render the text in a node. Do you have any suggestions for naming it to make it clearer? I can't come up with anything